cli/lib/kontena/cli/stacks/yaml/reader.rb

```diff
@@ -147,10 +157,18 @@ def process_variables
         raise RuntimeError, "Variable validation failed: #{variables.errors.inspect}" unless variables.valid?
       end
 
+      # @raise [Liquid::Error]
       def interpolate_liquid(content, vars)
         Liquid::Template.error_mode = :strict
         template = Liquid::Template.parse(content)
-        template.render(vars, strict_variables: true, strict_filters: true)
+
+        # Wrap nil values in LiquidNull to not have Liquid consider them as undefined
+        vars = vars.clone
```

Comment:
I'm not sure if it matters, but Hash#clone makes a shallow copy, meaning any hash/array values inside the top level hash are still references.